language/move-core/types/src/vm_status.rs

```diff
@@ -130,7 +130,7 @@ impl VMStatus {
                 let code = *code;
                 debug_assert!(code != StatusCode::EXECUTED);
                 debug_assert!(code != StatusCode::ABORTED);
-                debug_assert!(code.status_type() != StatusType::Execution);
+                debug_assert!(code.status_type() == StatusType::Execution);
```

Comment:
This debug assert is correct, because it should be an `ExecutionFailure` if it has this status type. Can you repro the failure?

Reply:
@tnowacki 
Please see https://github.com/move-language/move/issues/196
And regarding the place where VMStatus::Error(StatusCode::OUT_OF_GAS) generated, GasStatus::deduct_gas() will return PartialVMError::new(StatusCode::OUT_OF_GAS), and PartialVMError::finish(self, location: Location) -> VMError, and VMError::into_vm_status will return VMStatus::Error(major_status) other than VMStatus::ExecutionFailure if location is Location::Undefined. But most of the time VMRuntime::execute_function_impl() use Location::Undefined
Wish this can help